apis/python/tests/test_dense_nd_array.py

```diff
@@ -146,7 +168,12 @@ def test_dense_nd_array_slicing(tmp_path, io):
     SOMA's doubly-inclusive slice indexing semantics against Python's singly-inclusive slicing
     semantics, ensuring that none of the latter has crept into the former.
     """
-    a = soma.DenseNdArray(tmp_path.as_posix())
+    cfg = {}
```

Comment:
with all of the new error checks being performed in the read path, can we add some trivial tests that hit all of the exceptions/error checks?  Ie.,
* bogus coord type (eg., a string)
* bogus slice (eg,   slice(-1))
* points < 0, > size
etc

just want to make sure we cover the expected edge cases

Reply:
@bkmartinjr bogus slice, points < 0, > size, sure!

Bogus coord type, no we cannot because of your request on
https://github.com/single-cell-data/TileDB-SOMA/pull/581#discussion_r1041366130
namely typeguard _itself_ would fail our attempt to call with a type not matching the signature

Reply:
Our users do not run with typeguard, so I believe you still need to test it!

In other words, it is a weak unit test that only tests the "supproted" types!

Reply:
https://github.com/single-cell-data/TileDB-SOMA/issues/584